xds/src/main/java/io/grpc/xds/EnvoyProtoData.java

```diff
@@ -862,4 +1488,45 @@ static ClusterWeight fromEnvoyProtoClusterWeight(
       return new ClusterWeight(proto.getName(), proto.getWeight().getValue());
     }
   }
+
+  /**
+   * Convert protobuf message to human readable String format. Useful for protobuf messages
+   * containing {@link Any} fields.
+   */
+  @VisibleForTesting
+  static final class MessagePrinter {
```

Comment:
It makes no sense to put a message printer here. Only XdsClientImpl interacts with those proto messages directly.